nixos/modules/services/security/physlock.nix

```diff
@@ -105,7 +105,7 @@ in
         wantedBy = optional cfg.lockOn.suspend   "suspend.target"
                 ++ optional cfg.lockOn.hibernate "hibernate.target"
                 ++ cfg.lockOn.extraTargets;
-        before   = optional cfg.lockOn.suspend   "systemd-suspend.service"
+        before   = optionals cfg.lockOn.suspend  [ "systemd-suspend.service" "systemd-suspend-then-hibernate.service" ]
                 ++ optional cfg.lockOn.hibernate "systemd-hibernate.service"
```

Comment:
I think it should be added for `lockOn.hibernate` as well then

Reply:
Split it into a new `optional (suspend || hibernate)`.